4.cellprofiler_analysis/scripts/run_cellprofiler_pipe_with_sam_outputs.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib
+import pprint
+import shutil
+import sys
+
+sys.path.append("../../utils/")
+import cp_parallel
+
+# ## Set paths and variables
+
+# In[2]:
+
+
+# set the run type for the parallelization
+run_name = "analysis"
+
+# set main output dir for all plates
+output_dir = pathlib.Path("../analysis_output")
+output_dir.mkdir(exist_ok=True, parents=True)
+
+# directory where images are located within folders
+images_dir = pathlib.Path(
+    "../../2.cellprofiler_ic_processing/illum_directory"
+).resolve()
+# directory where masks are located within folders
+masks_dir = pathlib.Path("../../3b.run_sam/sam2_processing_dir/masks").resolve()
+
+# path to plugins directory as one of the pipelines uses the RunCellpose plugin
+plugins_dir = pathlib.Path(
+    "/home/lippincm/Documents/CellProfiler-plugins/active_plugins"
+).resolve()
+
+
+# In[3]:
+
+
+# make a new dir for input images
+CP_input_dir = pathlib.Path("../../3b.run_sam/sam2_processing_dir/CP_input/").resolve()
+CP_input_dir.mkdir(exist_ok=True, parents=True)
+
+# copy all images to the new dir
+for image in images_dir.rglob("*.tiff"):
+    if image.is_file():
+        shutil.copy(image, CP_input_dir)
+for mask in masks_dir.rglob("*.png"):
+    if mask.is_file():
+        shutil.copy(mask, CP_input_dir)
```

Comment:
Love this!

Reply:
Do you end up removing this temporary folder with the masks and tiffs after?

Reply:
No, not currently.